lib/fog/scaleway/compute.rb

```diff
@@ -150,7 +150,7 @@ def client
         end
 
         def endpoint
-          "https://cp-#{@region}.scaleway.com"
+          "https://api.scaleway.com/instance/v1/zones/#{@region}"
```

Comment:
I think the path in the URL will end up being overridden by the `:path` option given to `Fog::Core::Connection#request`.

We could use the [`:path_prefix`](https://github.com/fog/fog-core/blob/v1.45.0/lib/fog/core/connection.rb#L38) option in `Fog::Core::Connection#new` instead.

Something along the lines of this would probably work:

```diff
         def client
-          @client ||= Fog::Scaleway::Client.new(endpoint, @token, @connection_options)
+          @client ||= Fog::Scaleway::Client.new(endpoint, @token, connection_options)
         end

         def endpoint
-          "https://api.scaleway.com/instance/v1/zones/#{@region}"
+          'https://api.scaleway.com'
+        end
+
+        def connection_options
+          @connection_options.merge(path_prefix: path_prefix)
+        end
+
+        def path_prefix
+          "/instance/v1/zones/#{@region}"
         end
```

I guess we should also change the default zone to `fr-par-1`.

https://github.com/kaorimatz/fog-scaleway/blob/abf77115bc5c37fed0128586d216dc42428bdfc2/lib/fog/scaleway/compute.rb#L123

Reply:
It's been working fine for me from the vagrant provider, but I will follow you on this, it seems safer indeed.

Good catch about the default zone, I updated the README too. 🙂

Reply:
Great, thank you.

That's probably because https://api.scaleway.com/ works as an alias for https://api.scaleway.com/instance/v1/zones/fr-par-1/ at the moment:

```
$ curl -H 'X-Auth-Token: xxx' -v https://api.scaleway.com/servers
...
< warning: 299 - "Deprecated API please use https://api.scaleway.com/instance/v1/zones/fr-par-1/"
...
{"servers": []}
```

Just FYI, you can see what requests are being made by setting the EXCON_DEBUG environment variable.